### ISSUE TITLE
Feature Request: Expose CLIPGenericFlag sensors/switches to Home Assistant

### ISSUE THREAD

**Issue by snozzlebert** [Kane610/deconz#14]
Possible implementations:

- Expose the flag as a binary_sensor
or
- Expose them as a switch:
Map flag `true` to state 'on' and flag `false` to state 'off'
turn_on service to `PUT /sensors/#/state {"flag": true}`
turn_off service to `PUT /sensors/#/state {"flag": false}`

I would prefer the second one.

**Comment by Kane610** [comment]
What is the use case for this? From the look of it this will be a binary sensor. Could you post the json data for a generic flag?

**Comment by snozzlebert** [comment]
My use case is similar the the one described in this comment: https://github.com/dresden-elektronik/deconz-rest-plugin/issues/98#issuecomment-326562879
I'm using the rule engine of deCONZ and the CLIPGenericFlag is acting as a room's master switch.
I would like to see the state of this "master switch" (the flag) in Home Assistant.

Binary sensor would be ok for me.
What kind of JSON data do you need? 
My flag sensor:
```
{
        "config": {
            "on": true,
            "reachable": true
        },
        "modelid": "Switch",
        "name": "Kitchen Switch",
        "state": {
            "flag": true,
            "lastupdated": "2018-07-01T10:40:35"
        },
        "swversion": "1.0.0",
        "type": "CLIPGenericFlag",
        "uniqueid": "kitchen-switch"
    }
```

**Comment by Kane610** [comment]
Seems about right for the flag to be the main state. The Json data you gave me is good enough 👍

**Comment by Kane610** [comment]
Solved in v40.

PR is submitted to HASS, you can expect it to show up with release 0.74

**Comment by snozzlebert** [comment]
Thanks! One remark:
In the code is see:
```
GENERICFLAG = ['CLIPGenericFLAG']
```
shouldn't that be:
```
GENERICFLAG = ['CLIPGenericFlag']
```
if matching is case sensitive?
My JSON shows ``` "type": "CLIPGenericFlag",```

**Comment by Kane610** [comment]
You're right. Good catch!

**Comment by Kane610** [comment]
Updated PR

**Comment by snozzlebert** [comment]
I'm sorry to bother you again. I think I've found an issue in the GenericFlag class. The is_tripped function is missing. When I use the new code in Home Assistant I get:
```
  File "homeassistant\components\binary_sensor\deconz.py", line 71, in is_on
    return self._sensor.is_tripped
AttributeError: 'GenericFlag' object has no attribute 'is_tripped'
```

**Comment by Kane610** [comment]
Thanks for verifying, will fix immediately

**Comment by Kane610** [comment]
PR is updated

**Comment by Krocko** [comment]
Would it be possible to expose the CLIPGenericFlag as a switch to home assistant so we can toggle the flag easily in Home Assistant on and off?

**Comment by Kane610** [comment]
@Krocko can you create a new issue with examples on how it is set, and why it is useful

**Comment by Kane610** [comment]
https://github.com/ebaauw/homebridge-hue/wiki/Getting-Started#24-clip-sensors

**Comment by Krocko** [comment]
Hi @Kane610. Any news?

**Comment by Kane610** [comment]
I'm nearly done working on the UniFi integration, but I will put my time on the Axis integration prior to revisiting deconz

**Comment by Krocko** [comment]
Ok. Thank you for the info.

**Comment by Krocko** [comment]
Hello @Kane610 any update on this?

**Comment by Kane610** [comment]
No news, haven't investigated this at all

**Comment by Krocko** [comment]
Hi @Kane610. Do you have any news?

**Comment by Kane610** [comment]
Still waiting on examples

```
[Kane610](https://github.com/Kane610) commented [on 21 Dec 2019](https://github.com/Kane610/deconz/issues/14#issuecomment-568201742)
@Krocko can you create a new issue with examples on how it is set, and why it is useful
```

**Comment by Krocko** [comment]
Hello @Kane610 
Do you need more info?
